frontend/src/types/schemaSystem.ts

```diff
@@ -0,0 +1,478 @@
+import { SchemaReviewPolicyId } from "./id";
+import { Principal } from "./principal";
+
+// The engine type for rule template
+export type SchemaRuleEngineType = "MYSQL" | "COMMON";
+
+// The category type for rule template
+export type CategoryType = "ENGINE" | "NAMING" | "QUERY" | "TABLE" | "COLUMN";
+
+// The rule level
+export enum RuleLevel {
+  DISABLED = "DISABLED",
+  ERROR = "ERROR",
+  WARNING = "WARNING",
+}
+
+export const LEVEL_LIST = [
+  RuleLevel.ERROR,
+  RuleLevel.WARNING,
+  RuleLevel.DISABLED,
+];
+
+// The rule template payload
+enum PayloadType {
+  STRING = "STRING",
+  STRING_ARRAY = "STRING_ARRAY",
+  TEMPLATE = "TEMPLATE",
+}
+
+// StringPayload is the string type payload configuration options and default value.
+// Used by the frontend.
+interface StringPayload {
+  title: string;
+  description?: string;
```

Comment:
title and description can be put into RuleTemplatePayload?

Reply:
As we need to show different title (maybe also need desc) for different configurations in the frontend, I think we may need this to make it more clear for users.
For example, although the payload key for rule `naming.table` and rule `column.required` are all named as `format`, bu we need to use different title like "Column name format" and "Required column names". WDYT?

Reply:
Update to

```typescript
interface RuleTemplate {
  type: RuleType;
  category: CategoryType;
  engine: SchemaRuleEngineType;
  description: string;
  componentList: RuleComponent[];
  level: RuleLevel;
}
```

PTAL @tianzhou